src/lib/datetimeUtils.ts

```diff
@@ -28,23 +28,25 @@ export const idx2month = [
 	'December'
 ];
 
-export const monthIdxFromDateStr = (dateStr) => {
-	let mStr = /[0-9]+-([0-9]+)-[0-9]+/.exec(dateStr)[1];
-	return parseInt(mStr) - 1;
+export const monthIdxFromDateStr = (dateStr: string): number => {
+	const mdy = /[0-9]+-([0-9]+)-[0-9]+/.exec(dateStr);
```

Comment:
@getneil just one philosophical point here.  In this case, if mdy is null, that means there is a bug in the code that called this function because it should never be called with an arg that doesn't match the pattern.  By checking if mdy is null, this code might avoid crashing the program if there is a bug in the calling code, but it might also conceal that bug, making it more difficult to catch it and fix it.

Reply:
you are right. ill make it throw an error if null